activesupport/test/core_ext/duration_test.rb

```diff
@@ -559,7 +559,7 @@ def test_iso8601_parsing_wrong_patterns_with_raise
   def test_iso8601_output
     expectations = [
       ["P1Y",           1.year                           ],
-      ["P1W",           1.week                           ],
+      ["P7D",           1.week                           ],
```

Comment:
I think we should still expect `P1W` if there are only weeks in the duration, right? Try adding another test that better captures this edge case with weeks and other parts.